pandas/tests/indexes/multi/test_sorting.py

```diff
@@ -338,3 +338,12 @@ def test_sort_values_with_na_na_position(dtype, na_position):
         ]
     expected = MultiIndex.from_arrays(arrays)
     tm.assert_index_equal(result, expected)
+
+
+def test_sort_unnecessary_warning():
+    # GH#55386
+    midx = MultiIndex.from_tuples([(1.5, 2), (3.5, 3), (0, 1)])
+    midx = midx.set_levels([2.5, np.nan, 1], level=0)
+    result = midx.sort_values()
```

Comment:
Should we wrap this in a `tm.assert_produces_warning(False)`?

Reply:
Ci would raise anyway, that’s how I found this

Reply:
Ah okay, sounds good